src/solver/interpolate/bwd_bicubic_interpolate.cpp

```diff
@@ -0,0 +1,210 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/conv_solution.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/invoke_params.hpp"
+#include "miopen/miopen.h"
+#include <miopen/interpolate/solvers.hpp>
+#include <miopen/interpolate/utils.hpp>
+
+#include <miopen/interpolate/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/interpolate.hpp>
+#include <miopen/target_properties.hpp>
+
+#define LOCAL_SIZE_BWD_BICUBIC 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace interpolate {
+
+bool IsOverRocmBicubicBwd(const miopen::interpolate::BwdProblemDescription& problem)
+{
+    TensorDescriptor output_grad_desc = problem.GetOutputGradDesc();
+    TensorDescriptor input_grad_desc  = problem.GetInputGradDesc();
+    auto dtype                        = input_grad_desc.GetType();
+
+    float scale_h =
+        static_cast<float>(output_grad_desc.GetLengths()[2]) / input_grad_desc.GetLengths()[2];
+    float scale_w =
+        static_cast<float>(output_grad_desc.GetLengths()[3]) / input_grad_desc.GetLengths()[3];
+
+    if(dtype == miopenHalf || dtype == miopenBFloat16)
+    {
+        if(scale_h + scale_w < 8 && scale_h + scale_w > 1.4)
+            return true;
+        else
+            return false;
+    }
+    else
+    {
+        if(output_grad_desc.GetLengths()[2] + output_grad_desc.GetLengths()[3] <= 256 &&
+           (input_grad_desc.GetElementSize() >= 10000))
+            return true;
+        else
+            return false;
+    }
+}
+
+bool InterpolateBicubicBackward::IsApplicable(
+    const ExecutionContext&, const miopen::interpolate::BwdProblemDescription& problem) const
+{
+    if(problem.GetMode() != miopenInterpolateMode_t::MIOPEN_INTERPOLATE_MODE_BICUBIC)
+        return false;
+    if(!IsOverRocmBicubicBwd(problem))
+        return false;
+
+    return true;
+}
+
+ConvSolution InterpolateBicubicBackward::GetSolution(
+    const ExecutionContext& context,
+    const miopen::interpolate::BwdProblemDescription& problem) const
+{
+    std::ignore = context;
+
```

Comment:
you can add maybe_unused instead in the params. Ref: https://en.cppreference.com/w/cpp/language/attributes/maybe_unused